src/cryptography/x509.py

```diff
@@ -976,6 +982,11 @@ def __ne__(self, other):
         return not self == other
 
 
+class SubjectAlternativeName(GeneralNames):
```

Comment:
Ok, understood. Regarding the test cases, I'd assume we now have to run the
whole set of tests for SAN on the GeneralNames class as well.

Also, could you explain what the benefit of this solution is compared to
subclassing?

On Wed, May 27, 2015 at 8:15 AM, Alex Gaynor <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/1956#discussion_r31144337>:
>
> > @@ -976,6 +982,11 @@ def __ne__(self, other):
> >          return not self == other
> >
> >
> > +class SubjectAlternativeName(GeneralNames):
>
> You can probably just simplify __init__ to construct the GeneralNames, no
> need for the isinstance check, GeneralNames.__init__ will have that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1956/files#r31144337>.
>